iavl/iavl_proof2.go

```diff
@@ -0,0 +1,251 @@
+package iavl
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/tendermint/go-wire"
+)
+
+type PathToKey struct {
+	LeafHash   []byte
+	InnerNodes []IAVLProofInnerNode
+}
+
+func (p *PathToKey) String() string {
```

Comment:
More readable this way for sure.
Please add a comment on PathToKey to say the last InnerNodes is closest to root, this was confusing.